src/main/java/io/snyk/jenkins/SnykToHTML.java

```diff
@@ -3,61 +3,68 @@
 import hudson.EnvVars;
 import hudson.FilePath;
 import hudson.Launcher;
-import hudson.Util;
-import hudson.model.TaskListener;
 import hudson.util.ArgumentListBuilder;
 import io.snyk.jenkins.tools.SnykInstallation;
 import io.snyk.jenkins.transform.ReportConverter;
 import jenkins.model.Jenkins;
 
 import java.io.IOException;
+import java.time.Instant;
 
 import static io.snyk.jenkins.config.SnykConstants.SNYK_REPORT_HTML;
 import static io.snyk.jenkins.config.SnykConstants.SNYK_TEST_REPORT_JSON;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class SnykToHTML {
-  public static void generateReport(
+  public static FilePath generateReport(
     SnykContext context,
     SnykInstallation installation
-  ) throws IOException, InterruptedException {
-    TaskListener log = context.getTaskListener();
-    FilePath workspace = context.getWorkspace();
-    Launcher launcher = context.getLauncher();
-    EnvVars env = context.getEnvVars();
-
-    ArgumentListBuilder args = new ArgumentListBuilder();
-
-    FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
-    if (!snykReportJson.exists()) {
-      log.getLogger().println("Snyk report json doesn't exist");
-      return;
-    }
+  ) {
+    try {
+      FilePath workspace = context.getWorkspace();
+      Launcher launcher = context.getLauncher();
+      EnvVars env = context.getEnvVars();
+
+      FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
+      if (!snykReportJson.exists()) {
+        throw new RuntimeException("Snyk Report JSON does not exist.");
```

Comment:
The Report HTML is a hard requirement to progress to the archiving step. Previously we were just ignoring failed reports, but then it would fail during archiving as the HTML would not exist. Now I've changed it to fail early so it's clear what went wrong.